### PR TITLE
Multipleclustertests

### PR DESCRIPTION
**Pull request by sankarpn**
Adding Multiple clusters in domain controlled by operator.
1. testCreateDomainTwoConfiguredCluster - creates 2 configured cluster each having 2 managed servers
2. testCreateDomainTwoMixedCluster - creates a dynamic and a configured cluster

### CODE REVIEW

**Comment by vanajamukkara** [comment]
update usecases.md

File: integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    try {
+      Map<String, Object> domainMap = TestUtils.loadYaml(DOMAINONPV_WLST_YAML);
+      domainMap.put("domainUID", DOMAINUID);
+      domainMap.put("clusterType", "CONFIGURED");
+      domainMap.put(
+          "createDomainPyScript",
+          "integration-tests/src/test/resources/domain-home-on-pv/"
+              + TWO_CONFIGURED_CLUSTER_SCRIPT);
+
+      String add =
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: create a private method appendCluster2ToDomainTemplate() or some such name and call from both the tests

File: integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      logger.info("Making a backup of the domain template file:" + template);
+      if (!Files.exists(Paths.get(template + ".org"))) {
+        Files.copy(Paths.get(template), Paths.get(template + ".org"));
+      }
+      Files.write(Paths.get(template), add.getBytes(), StandardOpenOption.APPEND);
+      byte[] readAllBytes = Files.readAllBytes(Paths.get(template));
+      logger.info(new String(readAllBytes, StandardCharsets.UTF_8));
+      domain = TestUtils.createDomain(domainMap);
+      domain.verifyDomainCreated();
+      K8sTestUtils testUtil = new K8sTestUtils();
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: create a private method verifyDomain() and call from both the tests

File: integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        DOMAINUID + "-managed-server1",
+        DOMAINUID + "-managed-server2",
+        DOMAINUID + "-new-managed-server1",
+        DOMAINUID + "-new-managed-server2",
+      };
+      for (String pod : pods) {
+        assertTrue(
+            pod + " Pod not running", testUtil.isPodRunning(namespace, domain1LabelSelector, pod));
+      }
+      testBasicUseCases(domain);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: when a webapp is deployed, can you see to which cluster is targeting to ?

Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: It is deploying to the default cluster initialized in the domain cluster1

**Comment by sankarpn** [comment]
> 
> 
> update usecases.md

Done.

File: integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: Done

File: integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java (hunk 2 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: Done

File: integration-tests/USECASES.MD [hunk 4]
```diff
@@ -49,6 +49,8 @@ Basic Use Cases described above are verified in all the domain configurations. A
 | Domain with default sample values | create domain using mostly default values for inputs |	
 | Domain home in image using WLST | cluster scaling |
 | Domain home in image using WDT  | Configured cluster and cluster scaling |
+| Domain with 2 configured cluster created using WLST | Multi Configured cluster |
+| Domain with 2 mixed cluster created using WLST | 1 configured and 1 dynamic cluster in domain |
```
Comment (vanajamukkara) [inline integration-tests/USECASES.MD]: Domain with 2 clusters created using WLST

File: integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    logger.info("SUCCESS - " + testMethodName);
+  }
+
+  /**
+   * Call the basic usecases tests
+   *
+   * @param Domain
+   * @throws Exception
+   */
+  private void testBasicUseCases(Domain domain) throws Exception {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: move this method to BaseTest.java

File: integration-tests/USECASES.MD [hunk 6]
```diff
@@ -49,6 +49,8 @@ Basic Use Cases described above are verified in all the domain configurations. A
 | Domain with default sample values | create domain using mostly default values for inputs |	
 | Domain home in image using WLST | cluster scaling |
 | Domain home in image using WDT  | Configured cluster and cluster scaling |
+| Domain with 2 configured cluster created using WLST | Multi Configured cluster |
```
Comment (vanajamukkara) [inline integration-tests/USECASES.MD]: add under "Full test Configuration & Use Cases - Runs Quick test Configuration & Use cases and the below" in Domain Configuration and remove from here. 
```
Domain on PV with two configured clusters
Domain on PV with one configured cluster and one dynamic cluster
```

File: integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java (hunk 5 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/ITMultipleClusters.java]: Done

File: integration-tests/USECASES.MD (hunk 6 above)
Comment (sankarpn) [reply integration-tests/USECASES.MD]: Done

**Comment by sankarpn** [comment]
> 
> 
> LG, needs minor change. Update with jenkins results.

All tests passed in Jenkins.
http://wls-jenkins.us.oracle.com/view/weblogic-operator/job/weblogic-kubernetes-operator-javatest/1590/console

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Build Tools 1.0 .................................... SUCCESS [  1.348 s]
[INFO] weblogic-kubernetes-operator 2.2 ................... SUCCESS [  2.972 s]
[INFO] json-schema 2.2 .................................... SUCCESS [ 15.922 s]
[INFO] jsonschema-maven-plugin Maven Mojo 2.2 ............. SUCCESS [ 10.744 s]
[INFO] operator-model 2.2 ................................. SUCCESS [ 25.759 s]
[INFO] operator-swagger 2.2 ............................... SUCCESS [  5.270 s]
[INFO] operator-runtime 2.2 ............................... SUCCESS [01:02 min]
[INFO] operator-integration-tests 2.2 ..................... SUCCESS [  03:00 h]
[INFO] installation-tests 2.2 ............................. SUCCESS [  3.170 s]
[INFO] Project Reports 2.2 ................................ SUCCESS [  1.489 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:02 h
[INFO] Finished at: 2019-05-01T10:13:43-07:00

**Pull request merged**